fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -574,6 +574,50 @@ class ApplyMetadataMutationsImpl {
 		    .detail("LogRangeEnd", logRangeEnd);
 	}
 
+	void checkSetConstructKeys(MutationRef m) {
+		if (!SERVER_KNOBS->GENERATE_DATA_ENABLED) {
+			return;
+		}
+		if (!m.param1.startsWith(globalKeysPrefix)) {
+			return;
+		}
+		if (!toCommit) {
+			return;
+		}
+
+		if (m.param1.startsWith(constructDataKey)) {
+			// std::string s = "\xcf\xdf";
+			// Value v = encodeConstructValue(s, 100, 0x2ff, 123);
+			// TraceEvent(SevDebug,"ConstructDataDebug").detail("V", v);
+
+			uint64_t valSize, keyCount, seed;
+			Standalone<StringRef> prefix;
+			std::tie(prefix, valSize, keyCount, seed) = decodeConstructKeys(m.param2);
+			if (keyCount >= UINT16_MAX || valSize >= CLIENT_KNOBS->VALUE_SIZE_LIMIT) {
+				return;
+			}
+			uint8_t keyBuf[prefix.size() + sizeof(uint16_t)];
+			uint8_t* keyPos = prefix.copyTo(keyBuf);
+			*keyPos = '\xff';
+			StringRef keyEnd(keyBuf, keyPos - keyBuf + 1);
+			std::set<Tag> allTags;
+			for (auto it : keyInfo->intersectingRanges(KeyRangeRef(prefix, keyEnd))) {
```

Comment:
If i understand correctly, `prefixRange(prefix)` can be used instead of `KeyRangeRef(prefix, keyEnd)`.

Reply:
`prefixRange` expects a single `StringRef` to contain both the start and end key. But in this case, the start key (`prefix`) is in one `StringRef` and the end key (`keyEnd`) is a second `StringRef`. I could construct a single `StringRef` compatible with `prefixRange` but do not think that would save any work here.

Reply:
I see what you mean here. The storage server inserts keys with prefix, plus the little-endian bytes.